code/modules/events/ion_storm.dm

```diff
@@ -178,13 +179,29 @@ ABSTRACT_TYPE(/datum/ion_category)
 		if (!length(targets))
 			build_targets()
 		for (var/i in 1 to amount)
-			var/object = pick(targets)
+			var/atom/object = pick(targets)
+
+			//spatial interdictor: shield general hardware from ionic interference. law racks explicitly omitted due to sensitivity (and gameplay fun)
+			//consumes cell charge per hardware item protected, based on the category's interdict cost
+			var/interdicted = FALSE
+			for (var/obj/machinery/interdictor/IX in by_type[/obj/machinery/interdictor])
+				if (IN_RANGE(IX,object,IX.interdict_range) && IX.expend_interdict(interdict_cost))
+					interdicted = TRUE
+					SPAWN(rand(1,8))
```

Comment:
Is a spawn really necessary there? Could we just play the sound right away?

Reply:
Ionizations of each category happen in batches with no time gap between targets, making it so all the zaps happen at the same time; deferring them a bit gives more of the impression of the effect being deflected and dissipated.